kotlinx-coroutines-core/common/src/channels/BufferedChannel.kt

```diff
@@ -1690,8 +1690,12 @@ internal open class BufferedChannel<E>(
         }
 
         fun tryResumeHasNextOnClosedChannel() {
-            // Read the current continuation and clean
-            // the corresponding field to avoid memory leaks.
+            /*
```

Comment:
Please use `//` for comments in the code for consistency with the rest of the file.

Reply:
You mention multiple waiters in this comment, but it is always a single one. I suggest the following text instead; please feel free to modify it as you wish:

```
Read the current continuation of the suspended `hasNext()` call and clean the corresponding field to avoid memory leaks. While this nulling out is unnecessary, it eliminates memory leaks (through the continuation) if the channel iterator accidentally remains GC-reachable after the channel is closed. 
```